### PR TITLE
Feature/schedule overrides

### PR DESCRIPTION
**Pull request by emilssolmanis**
This looks scary, but it only really does 3 things:

* adds a `ScheduleMerger` interface and a (rather complicated) implementation. You might twist your brain trying to comprehend it, but it's 100% covered by tests for any possible edge case I could think\
 of. Suggestions for how to rewrite this better sorely needed, current consensus seems that we might be able to rewrite this with a 2-pass thing.
* adds an "override_sources" query arg to `ScheduleController` + implements relevant things in the parser and query object
* adds calls to the `ScheduleMerger` in the query executor

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/query/v4/schedule/EquivalentScheduleQueryExecutor.java [hunk 1]
```diff
@@ -83,20 +96,49 @@ public EquivalentScheduleResolverBackedScheduleQueryExecutor(
...
         if (query.isMultiChannel()) {
-            List<ChannelSchedule> channelSchedules = channelSchedules(schedule, query);
+            List<ChannelSchedule> channelSchedules;
+
+            if (query.getOverride().isPresent()) {
+                List<ChannelSchedule> originalSchedules = channelSchedules(schedule, query);
+                List<ChannelSchedule> overrideSchedules = channelSchedules(overrideSchedule, query);
+
+                channelSchedules = StreamUtils
+                        .zip(originalSchedules.stream(), overrideSchedules.stream(),
```
Comment (Dalamar42) [inline atlas-api/src/main/java/org/atlasapi/query/v4/schedule/EquivalentScheduleQueryExecutor.java]: Presumably the `originalSchedules` and the `overrideSchedules` will have the exact same schedules, in the same order and with no gaps, yes?

File: atlas-api/src/main/java/org/atlasapi/query/v4/schedule/ScheduleMergerImpl.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                            originalSchedule.getInterval().getEnd()
+                    ),
+                    mergedBuilder.build()
+            );
+        }
+
+        ItemAndBroadcast curr;
+        ItemAndBroadcast override = null;
+
+        boolean removeFollowOn = false;
```
Comment (Dalamar42) [inline atlas-api/src/main/java/org/atlasapi/query/v4/schedule/ScheduleMergerImpl.java]: Maybe set these as static finals rather than set them in the method itself

Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/query/v4/schedule/ScheduleMergerImpl.java]: Huh?

Comment (Dalamar42) [inline atlas-api/src/main/java/org/atlasapi/query/v4/schedule/ScheduleMergerImpl.java]: Actually nm. I think I misunderstood something. These are not flags to enable behaviour, but are rather tracking state during the merge?

File: atlas-api/src/main/java/org/atlasapi/query/v4/schedule/EquivalentScheduleQueryExecutor.java (hunk 1 above)
Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/query/v4/schedule/EquivalentScheduleQueryExecutor.java]: I should hope so. I should also probably throw if they don't.

File: atlas-api/src/main/java/org/atlasapi/query/v4/schedule/ScheduleMergerImpl.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+            removeFollowOn = false;
+            truncateFollowOnToOverride = false;
+
+            if (override == null) {
+                override = itOverride.next();
+            }
+
+            Broadcast overrideBroadcast = override.getBroadcast();
+
+            /* currStart >= overrideEnd, disjoint strictly after
```
Comment (Dalamar42) [inline atlas-api/src/main/java/org/atlasapi/query/v4/schedule/ScheduleMergerImpl.java]: Missing 'if'
